### PR TITLE
ui: tap experimental mode icon to toggle

### LINKED ISSUE #27002: Make experimental mode icon a toggle
Hit box should be a circle a bit bigger than the real icon; mis-taps are especially annoying because of the sidebar/nav/fullscreen cycling.

### PR DESCRIPTION
**Pull request by incognitojam**
After experimental mode has been enabled for the first time (confirmed), it can be toggled by tapping the engage-ability/experimental mode icon in the upper right.

Closes #27002

### CODE REVIEW

File: selfdrive/ui/qt/onroad.cc [hunk 1]
```diff
@@ -402,6 +402,21 @@ void AnnotatedCameraWidget::drawHud(QPainter &p) {
 // Window that shows camera view and variety of
 // info drawn on top
 
+void AnnotatedCameraWidget::mousePressEvent(QMouseEvent* e) {
+  const int exp_btn_w = radius + bdr_s * 4;
+  const int exp_btn_h = radius + bdr_s * 3;
+
+  if (engageable) {
+    auto params = Params();
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.cc:410]: keep this around as a class variable

File: selfdrive/ui/qt/onroad.cc [hunk 2]
```diff
@@ -402,6 +402,21 @@ void AnnotatedCameraWidget::drawHud(QPainter &p) {
...
+    if (params.getBool("ExperimentalModeConfirmed") && e->x() > rect().right() - exp_btn_w && e->y() < exp_btn_h) {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.cc:411]: not sure the rect is going to be intuitive when you mis-tap, think a circle would be better

File: selfdrive/controls/controlsd.py [hunk 3]
```diff
@@ -144,6 +144,7 @@ def __init__(self, sm=None, pm=None, can_sock=None, CI=None):
       self.params.remove("ExperimentalLongitudinalEnabled")
     if not self.CP.openpilotLongitudinalControl:
       self.params.remove("ExperimentalMode")
+      self.params.remove("ExperimentalModeConfirmed")
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:147]: why was this added?

Comment (incognitojam) [reply selfdrive/controls/controlsd.py:147]: Because I'm using `ExperimentalModeConfirmed` to determine whether experimental mode is available (not checking OP long/experimental long car params) If you moved from OP long to stock long you would still be able to toggle experimental mode onroad, unless this is cleared

Comment (adeebshihadeh) [reply selfdrive/controls/controlsd.py:147]: it should do the same checks as here: https://github.com/commaai/openpilot/blob/690999a2a6138368a93e818cbcebc984c7e58b91/selfdrive/ui/qt/offroad/settings.cc#L149

**Comment by adeebshihadeh** [comment]
This should also have pressed states, so probably best to just make this a real button.

File: selfdrive/ui/qt/onroad.cc (hunk 2 above)
Comment (sshane) [reply selfdrive/ui/qt/onroad.cc]: A circular tap area? I don't think you'd notice that much in the car.

**Comment by incognitojam** [comment]
Re-written using QPushButton with checked state when experimental long is enabled. Setting the opacity of the images works well as a pressed state

**Comment by incognitojam** [comment]
```
------------------------------------------------
------------ UI Draw Timing: master ------------
------------------------------------------------
min  1.89ms
max  19.87ms
std  1.47ms
mean 4.94ms
------------------------------------------------


------------------------------------------------
------------ UI Draw Timing: branch ------------
------------------------------------------------
min  2.04ms
max  16.12ms
std  1.09ms
mean 4.89ms
------------------------------------------------
```
